packages/nextjs/app/layout.tsx

```diff
@@ -4,9 +4,9 @@ import { ScaffoldEthAppWithProviders } from "~~/components/ScaffoldEthAppWithPro
 import { ThemeProvider } from "~~/components/ThemeProvider";
 import "~~/styles/globals.css";
 
-const baseUrl = process.env.NEXT_PUBLIC_VERCEL_URL
```

Comment:
using `VERCEL_URL` instead of `NEXT_PUBLIC_VERCEL_URL` to just follow convention what default [`metadataBase`](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#default-value) is following. 

Both [`VERCEL_URL`](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#:~:text=.-,VERCEL_URL,-The%20domain%20name) and [`NEXT_PUBLIC_VERCEL_URL`](https://vercel.com/docs/projects/environment-variables/system-environment-variables#framework-environment-variables) are prefilled by vercel so both works 🙌